aaudio/echo/src/main/cpp/echo_audio_engine.cc

```diff
@@ -123,11 +123,13 @@ void EchoAudioEngine::stopStreams() {
   if (recordingStream_ != nullptr) {
     stopStream(recordingStream_);
     closeStream(recordingStream_);
+    recordingStream_ = nullptr;
   }
 
   if (playStream_ != nullptr) {
     stopStream(playStream_);
     closeStream(playStream_);
+    playStream_ = nullptr;
```

Comment:
I see now that closeStream() also checks for null. So the destructor seems safe.